client/src/worldgen.rs

```diff
@@ -384,21 +423,51 @@ impl Surface {
     }
 }
 
-fn trilerp<N: na::RealField>(
+fn triserp<N: na::RealField>(
     &[v000, v001, v010, v011, v100, v101, v110, v111]: &[N; 8],
     t: na::Vector3<N>,
+    threshold1: N,
+    threshold2: N,
 ) -> N {
-    fn lerp<N: na::RealField>(v0: N, v1: N, t: N) -> N {
-        v0 * (N::one() - t) + v1 * t
+    fn serp<N: na::RealField>(v0: N, v1: N, t: N, threshold1: N, threshold2: N) -> N {
+        let out: N;
```

Comment:
You don't need an explicit result variable here, because if statements are expressions: just replace `out = whatever;` with `whatever` (note omitted semicolon)

Reply:
Made the change.